### ISSUE TITLE
Wilcoxon with Pratt behaves strangely in presence of ties

### ISSUE THREAD

**Issue by jlmeunier** [scipy/scipy#10223]
My issue is about Wilcoxon signed rank (scipy.stats.wilcoxon).

It offers 3 alternatives to deal with ties.

One is "pratt" and it behaves strangely: the pvalue goes to 0 when one 
adds ("enough") ties to a balanced number of wins and losses.

Since Pratt designed this method to deal with ties, I'm surprised by this
behavior.

Anyone more knowledgeable than me to tell if the implementation has an 
issue, or if this is what one must expect?

Thanks,

JL Meunier

### Reproducing code example:
```
import scipy.stats
print(scipy.stats.wilcoxon([1]*15 + [0]*40 + [-1]*15, zero_method="pratt"))
""" WilcoxonResult(statistic=832.5, pvalue=0.015400580278721392) """

""" should be close to 1.0, because both are clearly equal"""

"""like "zplitt" says:"""
print(scipy.stats.wilcoxon([1]*15 + [0]*40 + [-1]*15, zero_method="zsplit"))
""" WilcoxonResult(statistic=1242.5, pvalue=1.0) """

""" the number of ties is causing this problem:"""
print(scipy.stats.wilcoxon([1]*15          + [-1]*15, zero_method="pratt"))
""" WilcoxonResult(statistic=232.5, pvalue=1.0) """

print(scipy.stats.wilcoxon([1]*15 + [0]*34 + [-1]*15, zero_method="pratt"))
""" WilcoxonResult(statistic=742.5, pvalue=0.04390350653018313) """
```

### Scipy/Numpy/Python version information:
```
import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)

1.2.1 1.16.2 sys.version_info(major=3, minor=6, micro=8, releaselevel='final', serial=0)

```

**Comment by jlmeunier** [comment]
I tried it with R (first time using R...) and the results makes sense to me (in R , not in numpy)
```
x <- c(1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1)
y <- c(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
wilcox.test(x, y)

	Wilcoxon rank sum test with continuity correction

data:  x and y
W = 2450, p-value = 1
alternative hypothesis: true location shift is not equal to 0
```

**Comment by chrisb83** [comment]
Thanks for reporting this. I hope i fixed this in #10040 which is include in SciPy 1.3.0. can you test it in that version of SciPy?

**Comment by jlmeunier** [comment]
Hi,
Thanks!
Yes, now it Wilcoxon+Pratt does as expected. (tested with scipy 1.3.0 on win64)
